src/pages/filter/ui/panels/function-panel/function-panel.store.ts

```diff
@@ -95,6 +93,40 @@ class FunctionPanelStore {
   public fetchStatFunc(componentName: string, params: string) {
     return filterStore.fetchStatFuncAsync(componentName, params)
   }
+
+  public clearGroupFilter(): void {
+    const localSelectedFilters = cloneDeep(filterStore.selectedFilters)
+
+    if (localSelectedFilters[this.filterGroup]?.[this.filterName]) {
+      delete localSelectedFilters[this.filterGroup][this.filterName]
+    }
+
+    filterStore.setSelectedFilters(localSelectedFilters)
+  }
+
+  public get isFilterExistsInAppliedPreset(): boolean {
+    return (
+      datasetStore.activePreset !== '' &&
+      filterStore.selectedFilters[this.filterGroup]?.[this.filterName] !==
+        undefined
+    )
+  }
+
+  public isFilterInSelectedFilters(selectedFilterValue: string): boolean {
+    if (
+      filterStore.selectedFilters &&
```

Comment:
May be we can use optional chaining operator `?.` ?

e.g.

```
const selectedFilter = filterStore.selectedFilters?.[this.filterGroup]?.[this.filterName]
return selectedFilter && selectedFilterValue ? selectedFilter.hasOwnProperty(selectedFilterValue) : false
```

Reply:
Thanks, that looks better. However, using hasOwnProperty method is not recommended by eslint:
"Do not access Object.prototype method 'hasOwnProperty' from target object"